framework/src/util/CFRLogger.cpp

```diff
@@ -112,10 +115,38 @@ void CFRLogger::Log(const cppmicroservices::ServiceReferenceBase& sr,
                     const std::string& message,
                     const std::exception_ptr ex)
 {
-  auto currLogger = std::atomic_load(&logService);
-  if (currLogger) {
-    currLogger->Log(sr, level, message, ex);
+   if (IsOpen()) {
+    auto currLogger = std::atomic_load(&logService);
+    if (currLogger) {
+      currLogger->Log(sr, level, message, ex);
+    }
   }
 }
+
+void CFRLogger::Open()
+{
+  auto l = this->Lock();
+  US_UNUSED(l);
+  serviceTracker.reset(
+    new cppmicroservices::ServiceTracker<cppmicroservices::logservice::LogService> (GetBundleContext(), this));
```

Comment:
Why can't this be done in the constructor initializer lists as originally written?